### DISCUSSION TITLE
Unable to upload videos

### DISCUSSION THREAD

**Discussion by martinezhermes** [mediacms-io/mediacms#242] [Q&A]
I am trying to debug why is that I am unable to upload videos.

I have a docker deployed by default.
I access the server via my domain name using a nginx, I am unable to upload videos. Any media fails to upload. I am able to upload a profile picture though, all sites work I think? home keeps trying to load but since it is empty I guess it is fine? for it to continue to try?

I think the issue might be the proxy. I am unable to access the front via http://localhost because the computer server does not have an interface. I also configured the front domain variable to  mydomain.example.com (it is not actually that domain, just for example) in deploy/docker/local_settings.py

Here is my nginx:

```
upstream mydomain {
    server 192.168.0.10:80;
}
server {
    server_name mydomain.example.com;

    location / {
      proxy_pass http://mydomain;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;

    }

    listen [::]:443 ssl; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/mydomain.example.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/mydomain.example.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}

server {
    if ($host =  mydomain.example.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot
    listen 80;
    listen [::]:80;
    server_name  mydomain.example.com;
    return 404; # managed by Certbot
}
```

Any Idea on what I have to do to make it work?

**Comment by mgogoulos** [comment]
Can you write what docker-compose did you use, and what are the changes you did? So that we can reproduce the case

**Comment by martinezhermes** [comment]
Thanks for answering! they only line that I changed from the whole repo was `FRONTEND_HOST = 'https://mydomain.example.com/'` in `deploy/docker/local_settings.py` That's why I believe the issue has to do with the front not being able to push the upload to the back? due to some nginx configuration issue. I used the default compose`docker-compose.yaml` the issue that I have is that any file goes into retrying mode instantly, some cases it pushes X% of the file, say 5%? then it says retrying 1/2 ... 2/2 upload failed.

**Comment by mgogoulos** [comment]
mmm probably you have to use one of the other docker-compose files, as it's described on https://github.com/mediacms-io/mediacms/blob/main/docs/Docker_deployment.md

**Reply by swiftugandan** [reply]
1. The default docker-compose file is only applicable for http:// not https://
2. Is your DNS able to resolve mydomain.example.com ? If you are testing on the host machine, you may want to add an entry in your hosts file

**Reply by martinezhermes** [reply]
the thing is I am unable to reach the proxy of the customised compose from my own proxy, it says 503 service unavailable, I do resolve  mydomain.example.com correctly, I mean I can browse the whole site using the default compose, It is just the upload part that seems off. I'm still trying to figure out how to add the compose configuration to a previously defined proxy server that works for other services in the same cluster and has certificates.

**Reply by swiftugandan** [reply]
You might need to adjust the client max body size in your proxy.

See: https://github.com/mediacms-io/mediacms/blob/060bb457253063b8e56d492fdb99252dc4181131/deploy/docker/reverse_proxy/client_max_body_size.conf#L1

**Reply by martinezhermes** [reply]
> You might need to adjust the client max body size in your proxy.
> 
> See:
> 
> https://github.com/mediacms-io/mediacms/blob/060bb457253063b8e56d492fdb99252dc4181131/deploy/docker/reverse_proxy/client_max_body_size.conf#L1

this does in fact solve the uploading issue. However I am still unable to handle the process correctly. Getting a _Αn error occurred while loading the media's data_ message. I wonder if I need to write more endpoints (locations) into the configuration, I had some issues with other web apps in the past that required precisely defined endpoints in the nginx conf.

**Reply by swiftugandan** [reply]
https://github.com/mediacms-io/mediacms/issues/63#issuecomment-785086700 - i think you might fing this config useful
